File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/routeprogress/RouteProgressStateMap.java
package com.mapbox.services.android.navigation.v5.routeprogress;

import com.mapbox.navigator.RouteState;

import java.util.HashMap;

public final class RouteProgressStateMap extends HashMap<RouteState, RouteProgressState> {

  public RouteProgressStateMap() {
    put(RouteState.INVALID, RouteProgressState.ROUTE_INVALID);
    put(RouteState.INITIALIZED, RouteProgressState.ROUTE_INITIALIZED);
    put(RouteState.COMPLETE, RouteProgressState.ROUTE_ARRIVED);
    put(RouteState.TRACKING, RouteProgressState.LOCATION_TRACKING);
    put(RouteState.STALE, RouteProgressState.LOCATION_STALE);
    put(RouteState.OFFROUTE, null); // Ignore off-route (info already provided via listener)
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/routeprogress/RouteProgressStateMap.kt
package com.mapbox.services.android.navigation.v5.routeprogress

import com.mapbox.navigator.RouteState
import java.util.HashMap

class RouteProgressStateMap : HashMap<RouteState, RouteProgressState?>() {
    init {
        put(RouteState.INVALID, RouteProgressState.ROUTE_INVALID)
        put(RouteState.INITIALIZED, RouteProgressState.ROUTE_INITIALIZED)
        put(RouteState.COMPLETE, RouteProgressState.ROUTE_ARRIVED)
        put(RouteState.TRACKING, RouteProgressState.LOCATION_TRACKING)
        put(RouteState.STALE, RouteProgressState.LOCATION_STALE)
//        put(RouteState.UNCERTAIN, RouteProgressState.ROUTE_UNCERTAIN)
        put(RouteState.OFFROUTE, null) // Ignore off-route (info already provided via listener)
    }
}
